Sources/StreamChat/ChatClient.swift

```diff
@@ -49,206 +49,48 @@ public class ChatClient {
-
-    private(set) lazy var channelRepository = environment.channelRepositoryBuilder(
-        databaseContainer,
-        apiClient
-    )
-
-    private(set) lazy var messageRepository = environment.messageRepositoryBuilder(
-        databaseContainer,
-        apiClient
-    )
-
-    private(set) lazy var offlineRequestsRepository = environment.offlineRequestsRepositoryBuilder(
-        messageRepository,
-        databaseContainer,
-        apiClient
-    )
+    private(set) var offlineRequestsRepository: OfflineRequestsRepository
 
     /// A repository that handles all the executions needed to keep the Database in sync with remote.
-    private(set) lazy var syncRepository: SyncRepository = {
-        let channelRepository = ChannelListUpdater(database: databaseContainer, apiClient: apiClient)
-        return environment.syncRepositoryBuilder(
-            config,
-            activeChannelControllers,
-            activeChannelListControllers,
-            offlineRequestsRepository,
-            eventNotificationCenter,
-            databaseContainer,
-            apiClient
-        )
-    }()
+    private(set) var syncRepository: SyncRepository
 
     /// A repository that handles all the executions needed to keep the Database in sync with remote.
-    private(set) lazy var callRepository: CallRepository = {
-        environment.callRepositoryBuilder(apiClient)
-    }()
+    private(set) var callRepository: CallRepository
+
+    private(set) var channelRepository: ChannelRepository
 
     func makeMessagesPaginationStateHandler() -> MessagesPaginationStateHandling {
         MessagesPaginationStateHandler()
     }
 
     /// The `APIClient` instance `Client` uses to communicate with Stream REST API.
-    lazy var apiClient: APIClient = {
-        var encoder = environment.requestEncoderBuilder(config.baseURL.restAPIBaseURL, config.apiKey)
-        encoder.connectionDetailsProviderDelegate = self
-
-        let decoder = environment.requestDecoderBuilder()
-
-        let attachmentUploader = config.customAttachmentUploader ?? StreamAttachmentUploader(
-            cdnClient: config.customCDNClient ?? StreamCDNClient(
-                encoder: encoder,
-                decoder: decoder,
-                sessionConfiguration: urlSessionConfiguration
-            )
-        )
-
-        let apiClient = environment.apiClientBuilder(
-            urlSessionConfiguration,
-            encoder,
-            decoder,
-            attachmentUploader,
-            { [weak self] completion in
-                guard let self = self else {
-                    completion()
-                    return
-                }
-                self.refreshToken(
-                    completion: { _ in completion() }
-                )
-            },
-            { [weak self] endpoint in
-                self?.syncRepository.queueOfflineRequest(endpoint: endpoint)
-            }
-        )
-        return apiClient
-    }()
+    var apiClient: APIClient
```

Comment:
this one is both `var` and the setter is not private.

Reply:
The setter was not private before, I will need to investigate if this is user for testing or not 🤔